synapseclient/core/utils.py

```diff
@@ -342,12 +344,38 @@ def is_same_base_url(url1: str, url2: str) -> bool:
 def is_synapse_id_str(obj):
```

Comment:
I have not looked, but what functions are using this?  Would it make sense to update this function instead of adding another?